### PR TITLE
CI: Change build file names

### PR DESCRIPTION
**Pull request by cg2121**
### Description
This makes the build names with each OS consistent with each other.

### Motivation and Context
Makes it easier to work with them in different workflows.

### How Has This Been Tested?
Needs SEEKING_TESTERS tag to confirm.

### Types of changes
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by cg2121** [comment]
Confirmed it is working as expected.

**Comment by RytoEX** [comment]
> /CI/macos/03_package_obs.sh
> 
> > 141             FILE_NAME="obs-studio-mac0S-${VERSION_STRING}-x86_64.dmg"
> 
> obs-studio-mac"0"S-${VERSION_STRING}-x86_64.dmg" It is "0", not "O". Is this not a mistake?

That is a mistake. @cg2121 please fix that. Thanks.

**Comment by cg2121** [comment]
Fixed the typo

**Comment by WizardCM** [comment]
I would personally prefer we did the opposite - that all the zip files had capitalised "OBS Studio" but I'm not too fussed either way.

**Comment by cg2121** [comment]
Updated to latest master.

File: CI/linux/03_package_obs.sh [hunk 1]
```diff
@@ -43,14 +43,15 @@ package-obs-standalone() {
...
+    UBUNTU_VERSION=$(lsb_release -sr)
 
     if [ "${BUILD_FOR_DISTRIBUTION}" ]; then
         VERSION_STRING="${GIT_TAG}"
     else
         VERSION_STRING="${GIT_TAG}-${GIT_HASH}"
     fi
 
-    FILE_NAME="obs-studio-${VERSION_STRING}-Linux.deb"
+    FILE_NAME="obs-studio-ubuntu-${UBUNTU_VERSION}-${VERSION_STRING}.deb"
```
Comment (RytoEX) [inline CI/linux/03_package_obs.sh:54]: Prefer that the OBS Studio version string is immediately after "obs-studio" to associate the two pieces of information.

```
obs-studio-${VERSION_STRING}-ubuntu-${UBUNTU_VERSION}.deb
```

File: CI/macos/03_package_obs.sh [hunk 2]
```diff
@@ -134,11 +134,11 @@ package-obs-standalone() {
...
     if [ -z "${NOTARIZE_IMAGE}" -a -z "${NOTARIZE_BUNDLE}" ]; then
         if [ "${ARCH}" = "arm64" ]; then
-            FILE_NAME="obs-studio-${VERSION_STRING}-macOS-Apple.dmg"
+            FILE_NAME="obs-studio-macOS-${VERSION_STRING}-arm64.dmg"
         elif [ "${ARCH}" = "universal" ]; then
-            FILE_NAME="obs-studio-${VERSION_STRING}-macOS.dmg"
+            FILE_NAME="obs-studio-macOS-${VERSION_STRING}.dmg"
         else
-            FILE_NAME="obs-studio-${VERSION_STRING}-macOS-Intel.dmg"
+            FILE_NAME="obs-studio-macOS-${VERSION_STRING}-x86_64.dmg"
```
Comment (RytoEX) [inline CI/macos/03_package_obs.sh:141]: Prefer that the OBS Studio version string is immediately after "obs-studio" to associate the two pieces of information.
```
obs-studio-${VERSION_STRING}-macos-arm64.dmg
```
```
obs-studio-${VERSION_STRING}-macos.dmg
```
```
obs-studio-${VERSION_STRING}-macos-x86_64.dmg
```

File: CI/windows/03_package_obs.ps1 [hunk 3]
```diff
@@ -123,7 +123,7 @@ function Package-OBS-Standalone {
         $VersionString = "${GitTag}-${GitHash}"
     }
 
-    $FileName = "${ProductName}-${VersionString}"
+    $FileName = "obs-studio-windows-${VersionString}"
```
Comment (RytoEX) [inline CI/windows/03_package_obs.ps1:126]: Prefer that the OBS Studio version string is immediately after "obs-studio" to associate the two pieces of information.

```
obs-studio-${VERSION_STRING}-windows
```

**Review by RytoEX — CHANGES_REQUESTED** [review]
Discussed off-thread with @PatTheMav .

1. For the names of these intermediate and final artifacts, we'd prefer that "macos" is used.
2. For the names of the packages, we'd prefer that the OBS Studio version number immediately follows after the string "obs-studio" to closely associated the version number with the software name.

For example, for the package names (produced in the `03_package_obs.sh` scripts):
```
obs-studio-27.2.0-6ad63b94f-macos-arm64.dmg
obs-studio-27.2.0-6ad63b94f-macos-x86_64.dmg
obs-studio-27.2.0-ubuntu-20.04.deb
obs-studio-27.2.0-ubuntu-22.04.deb
obs-studio-27.2.0-6ad63b94f-windows-x64.zip
obs-studio-27.2.0-6ad63b94f-windows-x86.zip
```

You should probably also mirror the changes in the freebsd package script.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -193,7 +193,7 @@ jobs:
         if: ${{ success() && (github.event_name != 'pull_request' || env.SEEKING_TESTERS == '1') }}
         uses: actions/upload-artifact@v3
         with:
-          name: 'obs-macos-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
+          name: 'obs-studio-macOS-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
```
[the remark this answers is not in the document]
Comment (cg2121) [reply .github/workflows/main.yml]: Fixed

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -482,7 +482,7 @@ jobs:
         if: env.HAVE_CODESIGN_IDENTITY == 'true'
         uses: actions/download-artifact@v3
         with:
-          name: 'obs-macos-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
+          name: 'obs-studio-macOS-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
```
[the remark this answers is not in the document]
Comment (cg2121) [reply .github/workflows/main.yml]: Fixed

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -507,5 +507,5 @@ jobs:
         if: env.HAVE_CODESIGN_IDENTITY == 'true'
         uses: actions/upload-artifact@v3
         with:
-          name: 'obs-macos-${{ matrix.arch }}-notarized'
+          name: 'obs-studio-macOS-${{ matrix.arch }}-notarized'
```
[the remark this answers is not in the document]
Comment (cg2121) [reply .github/workflows/main.yml]: Fixed

File: CI/macos/03_package_obs.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (cg2121) [reply CI/macos/03_package_obs.sh]: fixed

Comment (cg2121) [reply CI/macos/03_package_obs.sh]: fixed

File: CI/windows/03_package_obs.ps1 (hunk 3 above)
Comment (cg2121) [reply CI/windows/03_package_obs.ps1]: fixed

File: CI/linux/03_package_obs.sh (hunk 1 above)
Comment (cg2121) [reply CI/linux/03_package_obs.sh]: fixed

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -348,14 +348,14 @@ jobs:
         if: ${{ success() && (github.event_name != 'pull_request' || env.SEEKING_TESTERS == '1') }}
         run: |
           CI/windows/03_package_obs.ps1 -BuildArch ${{ matrix.arch }} -Package
-          $ArtifactName = Get-ChildItem -filter "OBS-Studio-*-Win-${{ matrix.arch }}.zip" -File
+          $ArtifactName = Get-ChildItem -filter "obs-studio-windows-*-${{ matrix.arch }}.zip" -File
```
Comment (RytoEX) [inline .github/workflows/main.yml:351]: This line needs to be changed to:
```
$ArtifactName = Get-ChildItem -filter "obs-studio-*-windows-${{ matrix.arch }}.zip" -File
```

The pattern here comes from `DestinationPath` in `CI/windows/03_package_obs.ps1`:
```
    $FileName = "obs-studio-${VersionString}-windows"
```
```
            DestinationPath = "${FileName}-x64.zip"
```
So `DestinationPath` is: `obs-studio-${VersionString}-windows-${{ matrix.arch }}.zip`.

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -440,18 +440,18 @@ jobs:
...
 
-          Expand-Archive -Path "$(Get-ChildItem -filter "OBS-Studio-*-Win-x86.zip" -File)" -DestinationPath install_temp
-          Expand-Archive -Path "$(Get-ChildItem -filter "OBS-Studio-*-Win-x64.zip" -File)" -Force -DestinationPath install_temp
+          Expand-Archive -Path "$(Get-ChildItem -filter "obs-studio-windows-*-x86.zip" -File)" -DestinationPath install_temp
+          Expand-Archive -Path "$(Get-ChildItem -filter "obs-studio-windows-*-x64.zip" -File)" -Force -DestinationPath install_temp
 
           CI/windows/03_package_obs.ps1 -CombinedArchs -Package
 
-          $ArtifactName = (Get-ChildItem -filter "OBS-Studio-*-Win-x86+x64.zip" -File).Name
+          $ArtifactName = (Get-ChildItem -filter "obs-studio-windows-*-x86+x64.zip" -File).Name
```
Comment (RytoEX) [inline .github/workflows/main.yml:448]: Same as above.

This line needs to be changed to:
```
$ArtifactName = Get-ChildItem -filter "obs-studio-*-windows-${{ matrix.arch }}.zip" -File
```

The pattern here comes from `DestinationPath` in `CI/windows/03_package_obs.ps1`:
```
    $FileName = "obs-studio-${VersionString}-windows"
```
```
            DestinationPath = "${FileName}-x86+x64.zip"
```
So `DestinationPath` is: `obs-studio-${VersionString}-windows-x86+x64.zip`.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Requested changes to fix broken Windows jobs.

File: .github/workflows/main.yml (hunk 7 above)
Comment (cg2121) [reply .github/workflows/main.yml]: Fixed

File: .github/workflows/main.yml (hunk 8 above)
Comment (cg2121) [reply .github/workflows/main.yml]: Fixed

**Review by RytoEX — APPROVED** [review]
Seems fine to me now.

**Pull request merged**